package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html

```diff
@@ -64,6 +64,20 @@
 		return v and translate('enabled') or translate('disabled')
 	end
 
+	local function formatBits(v, factor)
+		--to format a value given in kilobits use a factor of 1000
```

Comment:
Isn't the value always in kilobits? Why not simplify the whole thing and call the function formatkilobits and drop the second parameter?

Reply:
I think could be simplified to (not tested):
```
	local function formatKbits(v)
		units = {'k', 'M', 'G'}
		pow = math.floor(math.log(v)/math.log(1000))
		return string.format('%s %sbit', v, units[pow])
	end
```

Reply:
This value is always kilobits, yes. I just didn't like to write the function less generic for an arbitrary factor.

Reply:
An unneeded feature in embedded development is called dead code :grin:...

Reply:
```
	local function formatKbits(v, factor)
		units = {'', 'k', 'M', 'G'}
                bits=v*(factor or 1)
		pow = math.floor(math.log(bits)/math.log(1000))
		return string.format('%s %sbit', bits/1000^pow, units[pow+1])
	end
```

That's the simplified version if you insist on keeping the feature I guess.

Reply:
I just see no point in showing 32342424242443232 Gbit/s

Reply:
Maybe then change the code to show the mantissa or something.

Reply:
Ah... I've found a solution for the display problem... Use "%g" instead of "%s". This forces the shortest notation.

Reply:
So it becomes:
```
	local function formatKbits(v, factor)
		units = {'', 'k', 'M', 'G'}
                bits=v*(factor or 1)
		pow = math.min(math.floor(math.log(bits)/math.log(1000)), #units-2)
		return string.format('%g %sbit', bits/1000^pow, units[pow+1])
	end
```

Reply:
I think you shouldn't handle negative values here, but with my proposed change of the if-clause in status-page.html directly... Maybe add an "else *UNLIMITED*"...

Reply:
@CodeFetch this is outdated, don't you agree?